diff --git a/src/Solver/MHDSolver_Compessible.jl b/src/Solver/MHDSolver_Compessible.jl
--- a/src/Solver/MHDSolver_Compessible.jl
+++ b/src/Solver/MHDSolver_Compessible.jl
@@ -8,14 +8,18 @@ export MHDcalcN_advection!
 using
   CUDA,
   TimerOutputs
 
+using FourierFlows: CPU
+
 include("MHDSolver.jl");
 include("HDSolver_Compessible.jl")
 ρUpdate! = HDSolver_compressible.ρUpdate!
 BᵢUpdate! = MHDSolver.BᵢUpdate!
 
 using LinearAlgebra: mul!, ldiv!
 
+synchronize_if_gpu(grid) = grid.device == CPU() ? nothing : CUDA.synchronize()
+
 # Definition of physical parameter between real space and spectral sapce
 # fft  - space parameters -> ρ px py pz bx by bz 
 # real - space parameters -> ρ ux uy uz bx by bz
@@ -92,44 +96,48 @@ function MHDcalcN_advection!(N, sol, t, clock, vars, params, grid)
 
 
-  @timeit_debug params.debugTimer "FFT Update" CUDA.@sync begin
+  @timeit_debug params.debugTimer "FFT Update" begin
     #Update ρ + P + V + B Real Conponment
     ldiv!(vars.ρ , grid.rfftplan, deepcopy(@view sol[:, :, :, params.ρ_ind ]))
     ldiv!(vars.ux, grid.rfftplan, deepcopy(@view sol[:, :, :, params.ux_ind]))
     ldiv!(vars.uy, grid.rfftplan, deepcopy(@view sol[:, :, :, params.uy_ind]))
     ldiv!(vars.uz, grid.rfftplan, deepcopy(@view sol[:, :, :, params.uz_ind]))
     ldiv!(vars.bx, grid.rfftplan, deepcopy(@view sol[:, :, :, params.bx_ind]))
     ldiv!(vars.by, grid.rfftplan, deepcopy(@view sol[:, :, :, params.by_ind]))
     ldiv!(vars.bz, grid.rfftplan, deepcopy(@view sol[:, :, :, params.bz_ind]))
 
     #Update momentum back to velocity
     @. vars.ux/=vars.ρ
     @. vars.uy/=vars.ρ
     @. vars.uz/=vars.ρ
 
 
     #Copy the spectral conponment to sketch array
     mul!(vars.uxh, grid.rfftplan, vars.ux)
     mul!(vars.uyh, grid.rfftplan, vars.uy)
     mul!(vars.uzh, grid.rfftplan, vars.uz)
+    synchronize_if_gpu(grid)
   end
   
   #Update continuity equation
-  @timeit_debug params.debugTimer "ρ Update" CUDA.@sync begin
+  @timeit_debug params.debugTimer "ρ Update" begin
     ρUpdate!(N, sol, t, clock, vars, params, grid)
+    synchronize_if_gpu(grid)
   end
 
   #Update V Advection
 
-  @timeit_debug params.debugTimer "UᵢUpdate" CUDA.@sync begin
+  @timeit_debug params.debugTimer "UᵢUpdate" begin
     UᵢUpdate!(N, sol, t, clock, vars, params, grid; direction="x")
     UᵢUpdate!(N, sol, t, clock, vars, params, grid; direction="y")
     UᵢUpdate!(N, sol, t, clock, vars, params, grid; direction="z")
+    synchronize_if_gpu(grid)
   end
   #Update B Advection
-  @timeit_debug params.debugTimer "BᵢUpdate" CUDA.@sync begin
+  @timeit_debug params.debugTimer "BᵢUpdate" begin
     BᵢUpdate!(N, sol, t, clock, vars, params, grid; direction="x")
     BᵢUpdate!(N, sol, t, clock, vars, params, grid; direction="y")
     BᵢUpdate!(N, sol, t, clock, vars, params, grid; direction="z") 
+    synchronize_if_gpu(grid)
   end
   return nothing
 end
